python/lsst/pipe/tasks/dcrModel.py

```diff
@@ -0,0 +1,472 @@
+        ------
+        ValueError
+            If neither ``modelImages`` or ``coaddExposure`` are set.
+            If ``modelImages`` is supplied but does not match ``dcrNumSubfilters``.
+        """
+        self.dcrNumSubfilters = dcrNumSubfilters
+        self.filterInfo = filterInfo
+        if modelImages is not None:
+            if len(modelImages) != dcrNumSubfilters:
+                raise ValueError("The dimension of modelImages must equal"
+                                 " the supplied dcrNumSubfilters.")
+            self.modelImages = modelImages
+        elif coaddExposure is not None:
+            maskedImage = coaddExposure.maskedImage.clone()
+            # NANs will potentially contaminate the entire image,
+            #  depending on the shift or convolution type used.
+            badPixels = np.isnan(maskedImage.image.array) | np.isnan(maskedImage.variance.array)
+            maskedImage.image.array[badPixels] = 0.
+            maskedImage.variance.array[badPixels] = 0.
+            maskedImage.image.array /= dcrNumSubfilters
+            maskedImage.variance.array /= dcrNumSubfilters
+            maskedImage.mask.array[badPixels] = maskedImage.mask.getPlaneBitMask("NO_DATA")
+            self.modelImages = [maskedImage, ]
+            for subfilter in range(1, dcrNumSubfilters):
+                self.modelImages.append(maskedImage.clone())
+        else:
+            raise ValueError("Either dcrModels or coaddExposure must be set.")
+
+    def __getitem__(self, subfilter):
+        """Iterate over the subfilters of the DCR model.
+
+        Parameters
+        ----------
+        subfilter : `int`
+            Index of the current subfilter within the full band.
+
+        Returns
+        -------
+        `lsst.afw.image.MaskedImageF`
+            The DCR model for the given ``subfilter``.
+        """
+        if subfilter >= self.dcrNumSubfilters:
+            raise IndexError("subfilter out of bounds.")
+        return self.modelImages[subfilter]
+
+    def getImage(self, subfilter, bbox=None):
+        """Return the image of the DCR model for a given subfilter.
+
+        Parameters
+        ----------
+        subfilter : `int`
+            Index of the current subfilter within the full band.
+        bbox : `lsst.afw.geom.box.Box2I`, optional
+            Sub-region of the coadd. Returns the entire image if None.
+
+        Returns
+        -------
+        `lsst.afw.image.MaskedImageF`
+            The DCR model for the given ``subfilter``.
+        """
+        return self.modelImages[subfilter] if bbox is None \
+            else self.modelImages[subfilter][bbox, afwImage.PARENT]
+
+    def getReferenceImage(self, bbox=None):
+        """Create a simple template from the DCR model.
+
+        Parameters
+        ----------
+        bbox : `lsst.afw.geom.box.Box2I`, optional
+            Sub-region of the coadd. Returns the entire image if None.
+
+        Returns
+        -------
+        `numpy.ndarray`
```

Comment:
Why a numpy array instead of an `Image`? It probably doesn't make much difference these days, but an `Image` is just a bit more capable.